mix/qml/js/ProjectModel.js

```diff
@@ -86,6 +85,7 @@ function saveProjectFile()
 		for (var i = 0; i < projectListModel.count; i++)
 			projectData.files.push(projectListModel.get(i).fileName);
 
+		projectSaving(projectData);
```

Comment:
This signal is used to notify all subscribers that the project is about to get saved, so that they could add custom information to saved data. Saving states is currently broken because of this